modules/cudev/include/opencv2/cudev/grid/reduce_to_vec.hpp

```diff
@@ -189,6 +189,7 @@ __host__ void gridReduceToColumn_(const SrcPtr& src, GpuMat_<ResType>& dst, cons
                                                                 shrinkPtr(mask),
                                                                 rows, cols,
                                                                 StreamAccessor::getStream(stream));
+    dst.reshape(dst.channels(), rows);
```

Comment:
`reshape` method creates new Mat object. You need to use something like `dst = dst.reshape(...);`.